contracts/orderValidation/ErrorCodes.md

```diff
@@ -0,0 +1,43 @@
+# Error Codes for invalidation
+
+## Overview
+
+- 000: Expected to be valid
+- 1\*\*: Nonce-related issues
+- 2\*\*: Amount-related issues
+- 3\*\*: Signature-related issues
+- 4\*\*: Whitelist-related issues
+- 5\*\*: Fee-related issues and other slippage
+- 6\*\*: Timestamp-related issues
+- 7\*\*: Transfer-related issues for ERC20/ERC721/ERC1155/non-compliant (approval and balances)
+
+## Details
+
+| Code | Description                                               |
+| ---- | --------------------------------------------------------- |
+| 000  | ORDER_EXPECTED_TO_BE_VALID                                |
+| 101  | NONCE_EXECUTED_OR_CANCELLED                               |
+| 102  | NONCE_BELOW_MIN_ORDER_NONCE                               |
+| 201  | ORDER_AMOUNT_CANNOT_BE_NULL                               |
+| 301  | MAKER_SIGNER_IS_NULL_SIGNER                               |
+| 302  | INVALID_S_PARAMETER_EOA                                   |
+| 303  | INVALID_V_PARAMETER_EOA                                   |
+| 304  | NULL_SIGNER_EOA                                           |
+| 305  | WRONG_SIGNER_EOA                                          |
+| 311  | SIGNATURE_INVALID_EIP1271                                 |
```

Comment:
let's make this ERC instead of EIP since you are using ERC below?

Reply:
Actually, EIP1271 is not an ERC (while the others are)

Reply:
Both OZ and Uniswap are using ERC

https://docs.openzeppelin.com/contracts/4.x/api/interfaces#IERC1271
https://docs.uniswap.org/protocol/reference/periphery/interfaces/external/IERC1271